protocol/encryption/protocol.go

```diff
@@ -477,12 +477,14 @@ func (p *Protocol) HandleMessage(
 		if dmProtocol == nil {
 			dmProtocol = encryptedMessage[noInstallationID]
 		}
-		hrHeader := dmProtocol.HRHeader
-		if hrHeader != nil && hrHeader.SeqNo == 0 {
-			// Payload contains hash ratchet key
-			err = p.encryptor.persistence.SaveHashRatchetKey(hrHeader.GroupId, hrHeader.KeyId, message)
-			if err != nil {
-				return nil, err
+		if dmProtocol != nil {
```

Comment:
it could be just else ?

Reply:
Not really, as `dmProtocol` can still be nil.

Reply:
There is an assignment in previous if.

Reply:
oh right